numpy/core/src/multiarray/alloc.c

```diff
@@ -113,8 +113,11 @@ npy_free_cache(void * p, npy_uintp sz)
 NPY_NO_EXPORT void *
 npy_alloc_cache_dim(npy_uintp sz)
 {
-    /* dims + strides */
-    if (NPY_UNLIKELY(sz < 2)) {
+    /*
+     * make sure any temporary allocation can be used for array metadata which
+     * uses one memory block for both dimensions and strides
+     */
```

Comment:
The most important reason for existence of the cache is the `npy_alloc_cache_dim(2 * nd)` allocation in `PyArray_NewFromDescr_int´. In the important cases (small arrays) nd is one.
To fill the cache with more values for arrays to use all other cache paths also allocate at least two entries.
It is probably not a particularly important optimization and could be removed with hardly an impact.